modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -172,6 +172,16 @@ class S3Repository extends MeteredBlobStoreRepository {
      */
     static final Setting<String> BASE_PATH_SETTING = Setting.simpleString("base_path");
 
+    /**
+     * The batch size for bulk deletion
+     */
+    static final Setting<Integer> DELETION_BATCH_SIZE_SETTING = Setting.intSetting(
+        "deletion_batch_size",
```

Comment:
Naming nit/suggestion: WDYT about `delete_objects_max_size`? The relevant S3 API is called [`DeleteObjects`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html) so I think mentioning that in the name would be helpful.

Reply:
Sounds good to me. Updated it as you suggested 61255e54b1b632193880817836a6cd960530b5de